modules/oe_theme_demo/src/Plugin/Styleguide/EclStyleGuide.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\oe_theme_demo\Plugin\Styleguide;
+
+use Drupal\styleguide\GeneratorInterface;
+use Drupal\styleguide\Plugin\StyleguidePluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\paragraphs\Entity\Paragraph;
+
+/**
+ * Comment Styleguide items implementation.
+ *
+ * @Plugin(
+ *   id = "ecl_styleguide",
+ *   label = @Translation("ECL Styleguide elements")
+ * )
+ */
+class EclStyleguide extends StyleguidePluginBase {
+
+  /**
+   * The styleguide generator service.
+   *
+   * @var \Drupal\styleguide\Generator
+   */
+  protected $generator;
+
+  /**
+   * The module handler service.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * Constructs a new CommentStyleguide.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\styleguide\GeneratorInterface $styleguide_generator
+   *   Styleguide generator service.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   Module handler service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, GeneratorInterface $styleguide_generator, ModuleHandlerInterface $module_handler) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->generator = $styleguide_generator;
+    $this->moduleHandler = $module_handler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('styleguide.generator'),
+      $container->get('module_handler')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function items() {
+    $items = [];
+    if ($this->moduleHandler->moduleExists('oe_paragraphs')) {
+
+      // Prepare quote paragraph.
+      $paragraph = Paragraph::create([
+        'type' => 'oe_quote',
+        'field_oe_text' => 'Quote author',
```

Comment:
The `$generator` service is used to generate random data. If we don't use it (one idea would be that we can't use it if we want to run visual regression on this page) then we have to remove it from the injected dependencies. Otherwise use here `$this->generator->sentence()` for example.